packages/forms-web-app/__tests__/unit/controllers/examination/file-upload/fileValidation.test.js

```diff
@@ -0,0 +1,233 @@
+const {
+	saveFileToDisk
+} = require('../../../../../src/controllers/examination/file-upload/fileManagement');
+const {
+	addFileToSession
+} = require('../../../../../src/controllers/examination/file-upload/fileSessionManagement');
+
+const {
+	noFileSelected,
+	moreThanXAmountFiles,
+	handleMultipleFileUploadsWithErrors
+} = require('../../../../../src/controllers/examination/file-upload/fileValidation');
+
+jest.mock('../../../../../src/controllers/examination/file-upload/fileManagement', () => ({
+	saveFileToDisk: jest.fn()
+}));
+jest.mock('../../../../../src/controllers/examination/file-upload/fileSessionManagement', () => ({
+	addFileToSession: jest.fn()
+}));
+
+describe('controllers/examination/file-upload/fileValidation', () => {
+	describe('#noFileSelected', () => {
+		describe('When a file has not been selected to upload', () => {
+			const files = [];
+			const result = noFileSelected(files);
+			it('should return the no file selected error', function () {
+				expect(result).toEqual({
+					errorMessage: {
+						text: 'Select a File'
+					},
+					errorSummary: [
+						{
+							text: 'Select a File'
+						}
+					]
+				});
+			});
+		});
+		describe('When a file has been selected to upload', () => {
+			const files = ['i am a file'];
+			const result = noFileSelected(files);
+			it('should return empty errors', function () {
+				expect(result).toEqual({
+					errorMessage: '',
+					errorSummary: []
+				});
+			});
+		});
+	});
+	describe('#moreThanXAmountFiles', () => {
+		describe('when the amount of files uploaded and in session', () => {
+			describe('are more than the total allowed', () => {
+				const filesToAdd = [21];
+				const currentFiles = [];
+				for (let i = 0; i < 20; i++) {
+					currentFiles.push(i);
+				}
+				const result = moreThanXAmountFiles(filesToAdd, currentFiles);
+				it('should then return the maximum file upload error', () => {
+					expect(result).toEqual({
+						errorMessage: {
+							text: 'You can only select up to 20 files at the same time'
+						},
+						errorSummary: [
+							{
+								text: 'You can only select up to 20 files at the same time'
+							}
+						]
+					});
+				});
+			});
+			describe('are less than the total allowed', () => {
+				const filesToAdd = [5];
+				const currentFiles = [1, 2, 3, 4];
+				const result = moreThanXAmountFiles(filesToAdd, currentFiles);
+				it('should return empty errors', () => {
+					expect(result).toEqual({
+						errorMessage: '',
+						errorSummary: []
+					});
+				});
+			});
+		});
+	});
+	describe('#handleMultipleFileUploadsWithErrors', () => {
+		describe('When an array of files could have file errors', () => {
```

Comment:
This covers all bad errors and i think is enough but i have covered them individually below can remove if this test is sufficient.